### DISCUSSION TITLE
Problem  BIRT v4.9 + TOMCAT v9/10

### DISCUSSION THREAD

**Discussion by AndrzejSulej** [eclipse-birt/birt#940] [Q&A] [answered]
Hi guys,

I try to change my current properly working version 4.8 of BIRT to 4.9

**Environment:**
Server: Ubuntu 20.04.1 LTS
Tomcat: 9.0.38
Java: OpenJDK 64-Bit Server VM (build 11.0.14.1 + 1-Ubuntu-0ubuntu1.20.04, mixed mode, sharing)
Birt-Runtime: https://download.eclipse.org/birt/downloads/drops/snapshot/birt-runtime-4.9.0-20220317.zip

Install path compatible with
https://eclipse.github.io/birt-website/docs/integrating/viewer-setup

After restart Tomcat
Page:
http://localhost:8080/manager/html
shows: 
My application path [/birt_4.9.315] status **Running: true**

Page:
http://localhost:8080/birt_4.9.317/
show:
```
Browser version: 4.9.0
Engine version: 4.9.0
JRE version: 11.0.14.1
```

After click: **View Example** : _(see response in attachment)_

What am I doing wrong or
forgot something?

Thanks for help.
[tomcat_response.txt](https://github.com/eclipse/birt/files/8500392/tomcat_response.txt)

**Comment by claesrosell** [comment]
This looks like a bug.
javax.xml.rpc_*.jar
should be part of the birt-runtime, but seems to be missing. If you want to you can download birt-runtime-osgi and copy the file from that zip file.

I will create an issue for this.

**Reply by AndrzejSulej** [reply]
I see 
javax.xml.rpc_1.1.2.v202203170910.jar
in 
/opt/tomcat/webapps/birt_4.9.317/WEB-INF/lib

**Comment by claesrosell** [comment]
Sorry, since that jar was missing from the "ReportEngine\lib" -folder I was assuming that it was completely missing from the distribution.
- My bad.
Still, that jar is the one that should contain the "javax.xml.rpc.handler.soap.SOAPMessageContext" class.

**Reply by AndrzejSulej** [reply]
I checked that, there is a SOAPMessageContext.class class in this jar [javax.xml.rpc_1.1.2.v202203170910.jar]

**Reply by claesrosell** [reply]
Yes, for some reason that jar seems to be missing from the classpath.

**Reply by AndrzejSulej** [reply]
Is any quick fix ?

**Reply by claesrosell** [reply]
I cannot promise anything but I will try to reproduce this problem on my Fedora machine during the coming days.

**Reply by AndrzejSulej** [reply]
I will be grateful

**Comment by AndrzejSulej** [comment]
I also tried on the v10.0.20 Tomcat.
Unfortunately, the problem is a bit deeper.

Status application Running:false

(see attachments)
[catalina.log](https://github.com/eclipse/birt/files/8500402/catalina.log)
[localhost.log](https://github.com/eclipse/birt/files/8500403/localhost.log)

**Reply by claesrosell** [reply]
Take it easy. Apache Tomcat v10 is part of the Jakarta EE transition (more info here: https://en.wikipedia.org/wiki/Jakarta_EE#History). Birt 4.9 will not work on Tomcat v10 but should work on Tomcat v9.

**Reply by AndrzejSulej** [reply]
it's clear

**Comment by wimjongman** [comment]
Please attach stack traces instead of pasting in a comment. The issue becomes very hard to read.

**Reply by AndrzejSulej** [reply]
done

**Comment by claesrosell** [comment, marked as answer]
I have now reproduced and found the problem. It is a packaing problem of "javax.xml.rpc", one of our osgi-ified dependencies. 
If you want test Birt 4.9 until a new version is release, this is how to fix the problem:

- Go to : "apache-tomcat-9.0.29/webapps/birt-viewer/WEB-INF/lib"
- Unzip the "javax.xml.rpc-api-1.1.2.jar" file that are _inside_ the javax.xml.rpc_1.1.2.v202203170910.jar
- Remove the javax.xml.rpc_1.1.2.v202203170910.jar
- Restart Tomcat

**Reply by AndrzejSulej** [reply]
👍 👍 👍 
It's working.
Thank you for your help.

Happy Easter Claes
